server/consumer.go

```diff
@@ -2247,16 +2247,21 @@ func (o *consumer) checkPendingRequests() {
 // This will release any pending pull requests if applicable.
 // Should be called only by the leader being deleted or stopped.
 // Lock should be held.
-func (o *consumer) releaseAnyPendingRequests() {
+func (o *consumer) releaseAnyPendingRequests(peerRemovalFlag bool) {
 	if o.mset == nil || o.outq == nil || o.waiting.len() == 0 {
 		return
 	}
-	hdr := []byte("NATS/1.0 409 Consumer Deleted\r\n\r\n")
+	var hdr []byte
+	if !peerRemovalFlag {
```

Comment:
This might want to be a deleted flag that scopes to the whole consumer. 

Also let's check that we do a stepdown first if we detect this consumer is the leader on a peer-remove and that process will send over the leader changed version.